Essentials/src/com/earth2me/essentials/utils/LocationUtil.java

```diff
@@ -175,6 +186,22 @@ public static Location getSafeDestination(final Location loc) throws Exception {
         int x = loc.getBlockX();
         int y = (int) Math.round(loc.getY());
         int z = loc.getBlockZ();
+        if (isBlockOutsideWorldBorder(world, x, z)) {
+            final Location center = world.getWorldBorder().getCenter();
+            final int radius = (int) world.getWorldBorder().getSize() / 2;
+            final int x1 = center.getBlockX() - radius, x2 = center.getBlockX() + radius;
+            final int z1 = center.getBlockZ() - radius, z2 = center.getBlockZ() + radius;
+            if (x < x1) {
+                x = x1;
+            } else if (x > x2) {
+                x = x2;
+            }
+            if (z < z1) {
+                z = z1;
+            } else if (z > z2) {
+                z = z2;
+            }
+        }
```

Comment:
After further review, this code should be placed after this line in AsyncTeleport so we do not create extraneous sync loads https://github.com/EssentialsX/Essentials/blob/2.x/Essentials/src/com/earth2me/essentials/AsyncTeleport.java#L181

Reply:
I still don't think this causes any sync loads. It's just getting the position of the block, not the block itself. Reducing sync loads in LocationUtil would probably be best for a separate PR, as that is not the concern of this one.